notebooks/ch12/linear_case_study.md

```diff
@@ -0,0 +1,1003 @@
+
+
+
+```python
+# HIDDEN
+sns.regplot('Height', 'Weight', donkeys_c);
+```
+
+
+![png](linear_case_study_files/linear_case_study_33_0.png)
+
+
+All three of our quantitative features have a linear relationship with our target variable of `Weight`, so we will not have to perform any transformations on our input data.
+
+It is also a good idea to see if our features are linear with each other. We plot two below: 
+
+
+```python
+# HIDDEN
+sns.regplot('Height', 'Length', donkeys_c);
+```
+
+
+![png](linear_case_study_files/linear_case_study_35_0.png)
+
+
+
+```python
+# HIDDEN
+sns.regplot('Height', 'Girth', donkeys_c);
+```
+
+
+![png](linear_case_study_files/linear_case_study_36_0.png)
+
+
+From these plots, we can see that our predictor variables also have strong linear relationships with each other. This makes our model harder to interpret, so we should keep this in mind after we create our model.
+
+## Transforming Variables
+
+Next, we transform the categorical variables into dummy variables so that we can include them in the model. Recall from our boxplots that `Sex` was not a useful variable, so we will drop it. We will also remove the `WeightAlt` column because we only have its value for 31 donkeys.
+
+
+```python
+donkeys_c = donkeys_c.drop(['Sex', 'WeightAlt'], axis=1)
+donkeys_c = pd.get_dummies(donkeys_c, columns=['BCS', 'Age'])
+donkeys_c.head()
+```
+
+
+
+
+<div>
+<style>
+    .dataframe thead tr:only-child th {
+        text-align: right;
+    }
+
+    .dataframe thead th {
+        text-align: left;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>Length</th>
+      <th>Girth</th>
+      <th>Height</th>
+      <th>...</th>
+      <th>Age_5-10</th>
+      <th>Age_&lt;2</th>
+      <th>Age_&gt;20</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>78</td>
+      <td>90</td>
+      <td>90</td>
+      <td>...</td>
+      <td>0</td>
+      <td>1</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>91</td>
+      <td>97</td>
+      <td>94</td>
+      <td>...</td>
+      <td>0</td>
+      <td>1</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>74</td>
+      <td>93</td>
+      <td>95</td>
+      <td>...</td>
+      <td>0</td>
+      <td>1</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>87</td>
+      <td>109</td>
+      <td>96</td>
+      <td>...</td>
+      <td>0</td>
+      <td>1</td>
+      <td>0</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>79</td>
+      <td>98</td>
+      <td>91</td>
+      <td>...</td>
+      <td>0</td>
+      <td>1</td>
+      <td>0</td>
+    </tr>
+  </tbody>
+</table>
+<p>5 rows × 16 columns</p>
+</div>
+
+
+
+Since we do not want our matrix to be over-parameterized, we should drop one category from the `BCS` and `Age` dummies.
```

Comment:
wait omg is this a thing LOL

I'm guessing you based this off Prof. Nolan's notebook, so in that case, 🙌  but I think something that makes mores sense to me is merging certain categories of the same variable if we notice that the granularity is unnecessary

Reply:
it is a thing! say you have a cat who is either cute or not-cute (only one of them!). if you have a column for both cute and not-cute, that is unnecessary because if cute is 1, you know that not-cute is 0, etc.